src/components/dom.jsx

```diff
@@ -59,6 +60,10 @@ const DOM = (props) => {
 		escapedState,
 	};
 
+	if(clientFilename){
```

Comment:
Can you elaborate? It's hard to evaluate this kind of change without a description of what it's moving toward - as a standalone change, it is a step backward because it allows the client to call `makeRenderer` in a way that would silently break the application (no client-side rendering).

If this change is only needed when we have a different implementation to replace it, I'd prefer to make the change at that point to see it in context.